cmd/lk/agent.go

```diff
@@ -785,27 +785,25 @@ func deleteAgent(ctx context.Context, cmd *cli.Command) error {
 	}
 
 	var res *lkproto.DeleteAgentResponse
-	var innerErr error
+	var agentErr error
 	if err := util.Await(
 		"Deleting agent ["+util.Accented(agentID)+"]",
 		func() {
-			if res, innerErr = agentsClient.DeleteAgent(ctx, &lkproto.DeleteAgentRequest{
+			res, agentErr = agentsClient.DeleteAgent(ctx, &lkproto.DeleteAgentRequest{
 				AgentId: agentID,
-			}); err != nil {
-
-			}
+			})
 		},
 	); err != nil {
 		return err
 	}
 
-	if innerErr != nil {
-		if twerr, ok := err.(twirp.Error); ok {
+	if agentErr != nil {
+		if twerr, ok := agentErr.(twirp.Error); ok {
 			if twerr.Code() == twirp.PermissionDenied {
 				return fmt.Errorf("agent hosting is disabled for this project -- join the beta program here [%s]", cloudAgentsBetaSignupURL)
 			}
 		}
-		return err
+		return agentErr
```

Comment:
In this case should we maybe not return the error at all? Below we check `res.Success`, and maybe its okay to swallow an error if it was not catastrophic? Unsure if that can even happen, but just thinking out loud. We don't want to lead user to believe the agent wasn't deleted if it actually succeeded.

Reply:
Looking at the server code, the error always indicates that the agent was not deleted. Not sure why the error was swallowed earlier.